src/server/task_manager/client_wrapper.ts

```diff
@@ -32,9 +46,26 @@ export class TaskManagerClientWrapper {
   }
 
   public schedule(task: TaskInstance) {
-    if (this.client == null) {
-      throw new Error('Task Manager Client has not been set properly!');
+    const { error } = checkClient(this.client);
```

Comment:
I think instead of putting this check in each method, we should just have a sane default client, and put this check in any place that sets the client.